security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,207 @@
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end test ensuring no requests are included in blocks
+      - https://github.com/ethereum-optimism/optimism/pull/14253
+  2. Simplicity of validation rules - always must be an empty hash
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing requests hash logic in respective clients (e.g. `op-node`, `op-geth`, `op-reth`, ...)
+
+### FM3: EIP-7702 transactions cannot be included on the chain
+
+- **Description:** A client implementation could have a bug that disallows SetCode (EIP-7702) transactions from being included on chain. This could cause an unintended fork by the affected client.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:** 
+	1. End-to-end tests of set code transactions before and after the fork
+        - https://github.com/ethereum-optimism/optimism/pull/14197
+        - https://github.com/ethereum-optimism/optimism/pull/14288
+- **Detection:** Sending set code transactions would fail - manual detection is most likely.
+- **Recovery Path(s)**: This would require an emergency client update of the implementation with the bug (e.g. `op-reth`, `op-geth`, ...).
+
+### FM4: BLS Precompiles could cause increased FP Program execution time
+
+- **Description:** If implemented in the FP program, BLS precompiles could greatly increase computation time.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program.
+        - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- **Detection:** op-program trace runner computation time would increase significantly (should trigger an alert)
+- **Recovery Path(s)**: If this occurred, which it can't with the mitigations, we'd have to make op-program efficient enough to execute
+BLS precompiles in sufficient time.
+
+### FM5: BLS Precompiles could fail to execute in the FP program
+
+- **Description:** If implemented in the FP program, BLS precompiles could fail to execute. This could cause certain blocks to be unprovable.
+- **Risk Assessment:** High impact, low likelihood
+- **Mitigations:**
+	1. Fully mitigated by using an accelerated precompile that calls out to L1 instead of calculating inside the program. This means it should 
+  have the same performance characteristics as existing crypto precompiles like `ecrecover`.
+        - [Specs](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- **Detection:** op-program trace runner failure (possibly triggering an alert)
```

Comment:
Can we be more specific about the kind of alert that would be triggered by the `vm-runner` here? 
Probably this would be OOM, If this the performance, I assume?

Reply:
See below

Reply:
Since this would be similar to the previous Failure mode, here we move forward.